src/movies.js

```diff
@@ -1,23 +1,112 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+  // return todos diretores
+  const allDirectors = moviesArray.map((movie) => {
+    return movie.director;
+  });
+  return allDirectors;
+}
+function getSingleDirectors(moviesArray) {
+  return [...new Set(getAllDirectors(moviesArray))];
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+//apenas filmes de Drama
+function howManyMovies(moviesArray) {
+  let filteredDrama = [];
+  moviesArray.filter((eachMovie) => {
+    // include procura dentro do array se há a classificação de drama
+    if (
+      eachMovie.genre.includes("Drama") &&
+      eachMovie.director.includes("Steven Spielberg")
+    ) {
+      filteredDrama.push(eachMovie);
+    }
+  });
+  return filteredDrama.length;
+}
+
+function howManyDramaMovies(moviesArray) {
+  let filteredDrama = [];
+  moviesArray.filter((eachMovie) => {
+    if (eachMovie.genre.includes("Drama")) {
+      filteredDrama.push(eachMovie);
+    }
+  });
+  return filteredDrama;
+}
 
+// genres:
+// crime, drama, action, thriller, biography, history, adventure, horror, music
+// fantasy, western, comedy, romance, sci-fi, mystery, family, war, animation, film-noir
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+  if (!moviesArray.length) {
+    return 0;
+  }
+  // calcula score
+  let mediaScore = moviesArray.reduce((acc, eachMovie) => {
+    if (!eachMovie.score) {
+      eachMovie.score = 0;
+    }
+    return acc + eachMovie.score;
+  }, 0);
+  // calcula media score
+  let mediaMovie = mediaScore / moviesArray.length;
+  //return a media em duas casas
+  return Number(parseFloat(mediaMovie).toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+// apenas Dramas
+function dramaMoviesScore(moviesArray) {
+  const dramaMovies = moviesArray.filter((eachMovie) => {
+    return eachMovie.genre.includes("Drama");
+  });
+  if (!dramaMovies.length) {
+    return 0;
+  }
```

Comment:
eu só colocaria esse `if` no começo, pois se ele for vazio não preciso fazer o filtro antes, certo?